dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -1512,6 +1529,349 @@ class FunctionStringRegexpInstr : public FunctionStringRegexpBase
+        {
+            // Index at 0, 1 and 4 arguments should be string type, otherwise int type.
+            is_str_arg = (i <= 1 || i == 4);
+            checkInputArg(arguments[i], is_str_arg, &has_nullable_col, &has_data_type_nothing);
+        }
+
+        if (has_data_type_nothing)
+            return std::make_shared<DataTypeNullable>(std::make_shared<DataTypeNothing>());
+
+        return std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>());
+    }
+
+    template <typename ExprT, typename PatT, typename PosT, typename OccurT, typename MatchTypeT>
+    void REGEXP_CLASS_MEM_FUNC_IMPL_NAME(ColumnWithTypeAndName & res_arg, const ExprT & expr_param, const PatT & pat_param, const PosT & pos_param, const OccurT & occur_param, const MatchTypeT & match_type_param) const
+    {
+        size_t col_size = expr_param.getDataNum();
+
+        // Get function pointers to process the specific int type
+        GetIntFuncPointerType get_pos_func = getGetIntFuncPointer(pos_param.getIntType());
+        GetIntFuncPointerType get_occur_func = getGetIntFuncPointer(occur_param.getIntType());
+
+        // Container will not be used when parm is const
+        const void * pos_container = pos_param.getContainer();
+        const void * occur_container = occur_param.getContainer();
+
+        // Const value will not be used when param is not const
+        Int64 pos_const_val = PosT::isConst() ? pos_param.template getInt<Int64>(0) : -1;
+        Int64 occur_const_val = OccurT::isConst() ? occur_param.template getInt<Int64>(0) : -1;
+
+        // Check if args are all const columns
+        if constexpr (ExprT::isConst() && PatT::isConst() && PosT::isConst() && OccurT::isConst() && MatchTypeT::isConst())
+        {
+            if (expr_param.isNullAt(0) || pat_param.isNullAt(0) || pos_param.isNullAt(0) || occur_param.isNullAt(0) || match_type_param.isNullAt(0))
+            {
+                res_arg.column = res_arg.type->createColumnConst(col_size, Null());
+                return;
+            }
+
+            String pat = pat_param.getString(0);
+            if (unlikely(pat.empty()))
+                throw Exception(EMPTY_PAT_ERR_MSG);
+
+            int flags = getDefaultFlags();
+            String expr = expr_param.getString(0);
+            String match_type = match_type_param.getString(0);
+            pat = fmt::format("({})", pat);
+
+            Regexps::Regexp regexp(addMatchTypeForPattern(pat, match_type, collator), flags);
+            auto res = regexp.substr(expr.c_str(), expr.size(), pos_const_val, occur_const_val);
+            if (res)
+                res_arg.column = res_arg.type->createColumnConst(col_size, toField(res.value().toString()));
+            else
+                res_arg.column = res_arg.type->createColumnConst(col_size, Null());
+            return;
+        }
+
+        // Initialize result column
+        auto col_res = ColumnString::create();
+        col_res->reserve(col_size * 15);
```

Comment:
Why reserve `col_size * 15`? The input of `reserve` is the element size you want to reserve.

Reply:
> Why reserve `col_size * 15`? The input of `reserve` is the element size you want to reserve.

I consider it as the byte size at the beginning. I will modify it.